Trimble.Modus.Components/Controls/TMRadioButton/TMRadioButtonGroup.cs

```diff
@@ -0,0 +1,501 @@
+        if (_radioButtons.Count == SelectedIndex + 1)
+        {
+            radioButton.IsSelected = true;
+        }
+        radioButton.IsEnabled = IsEnabled;
+        radioButton.Size = Size;
+        radioButton.SelectionChanged += RadioButtonSelectionChanged;
+    }
+
+    /// <summary>
+    /// This method is called when a radio button is selected.
+    /// </summary>
+    /// <param name="sender"></param>
+    /// <param name="e"></param>
+    private void RadioButtonSelectionChanged(object sender, EventArgs e)
+    {
+        if (sender is not TMRadioButton selectedRadioButton || !selectedRadioButton.IsSelected)
+        {
+            return;
+        }
+
+        foreach (var radioButton in _radioButtons)
+        {
+            if (radioButton != selectedRadioButton)
+            {
+                radioButton.IsSelected = false;
+            }
+        }
+
+        int index = FindIndex(selectedRadioButton);
+        SelectedIndex = index;
+        object value = selectedRadioButton.Value;
+        var eventArgs = new TMRadioButtonEventArgs(value, index);
+        SelectedRadioButtonChangedCommand?.Execute(eventArgs);
+        SelectedRadioButtonChanged?.Invoke(this, eventArgs);
+    }
+
+    /// <summary>
+    /// Finds the index of the selected radio button in the group
+    /// </summary>
+    /// <param name="radioButton">Selected Radio button</param>
+    /// <returns>integer index</returns>
+    private int FindIndex(TMRadioButton radioButton)
+    {
+        int index = 0;
+
+        return FindIndexRecursive(_buttonContainer.Children, radioButton, index);
+    }
+
+    /// <summary>
+    /// Finds the index of the selected radio button in the group
+    /// </summary>
+    /// <param name="children">List of children's in <see cref="TMRadioButtonGroup"/></param>
+    private int FindIndexRecursive(IList<IView> children, TMRadioButton radioButton, int index)
+    {
+        foreach (var child in children)
+        {
+            if (child is TMRadioButton)
+            {
+                if (child == radioButton)
+                {
+                    return index;
+                }
+                index++;
+            }
+            else if (child is Layout childLayout)
+            {
+                int childIndex = FindIndexRecursive(childLayout.Children, radioButton, index);
+                if (childIndex != -1)
+                {
+                    return childIndex;
+                }
+            }
+        }
+
+        return -1;
+    }
+
+    /// <summary>
+    /// Loops through all the children of the radio button group
+    /// and sets the IsSelected property of the radio buttons in the group
+    /// </summary>
+    /// <param name="children">List of children's in <see cref="TMRadioButtonGroup"/> </param>
+    /// <param name="index">Index of the selected <see cref="TMRadioButton"/></param>
+    private void LoopChildren(IList<IView> children, int index)
```

Comment:
This method, not only loops the children, but also changes it's IsSelectted property. Can you rename this method to `SelectChildrenAtIndex` or something like that?

Reply:
Done